lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb

```diff
@@ -137,6 +137,43 @@ def create_release(repository:, version:, description:, assets:, prerelease:, is
         release[:html_url]
       end
 
+      # Use the GitHub API to generate release notes based on the list of PRs between current tag and previous tag.
+      # @note This API uses the `.github/release.yml` config file to classify the PRs by category in the generated list according to PR labels.
+      #
+      # @param [String] repository The repository to create the GitHub release on. Typically a repo slug (<org>/<repo>).
+      # @param [String] tag_name The name of the git tag to generate the changelog for.
+      # @param [String] previous_tag The name of the git tag to compare to.
+      # @param [String] target_commitish The commit sha1 or branch name to use as the head for the comparison if the `tag_name` tag does not exist yet. Unused if `tag_name` exists.
+      # @param [String] config_file_path The path to the GitHub configuration file to use for generating release notes. Will use `.github/release.yml` by default if it exists.
+      #
+      # @return [String] The string returned by GitHub API listing PRs between `previous_tag` and current `tag_name`
+      # @raise [StandardError] Might raise if there was an error during the API call
+      #
+      def generate_release_notes(repository:, tag_name:, previous_tag:, target_commitish: nil, config_file_path: nil)
+        repo_path = Octokit::Repository.path(repository)
+        api_url = "#{repo_path}/releases/generate-notes"
```

Comment:
TIL that there's such thing in the API, nice!

Reply:
Yep, very useful one, and thankfully… because without it, it would have been quite a nightmare to correctly implement a similar automatic logic ourselves, especially for cases of complicated git tree graphs (see paaHJt-4Ne-p2).

Strangely there's no built-in shim/wrapper method [in the Octokit ruby Client](https://octokit.github.io/octokit.rb/Octokit/Client/Releases.html) for this API endpoint, hence us having to call the low-level `client.post(…)` method ourselves to use it 🙃  eh. 🤷